packages/server/src/integrations/oracle.ts

```diff
@@ -63,22 +68,250 @@ module OracleModule {
+              name: tableName,
+              columns: {}
+            }
+            oracleTables[tableName] = table
+          }
+
+          let column = table.columns[columnName]
+          if (!column) {
+            column = {
+              name: columnName,
+              type: dataType,
+              default: dataDefault, 
+              id: columnId,
+              constraints: {}
+            }
+            table.columns[columnName] = column
+          }
+
+          if (constraintName && constraintType) {
+            let constraint = column.constraints[constraintName]
+            if (!constraint) {
+              constraint = {
+                name: constraintName,
+                type: constraintType,
+                relatedConstraintName: relatedConstraintName,
+                searchCondition: searchCondition
+              }
+            }
+            column.constraints[constraintName] = constraint
+          }
+        })
+      }
+
+      return oracleTables
+    }
+
+    private isSupportedColumn(column: OracleColumn) {
+      if (UNSUPPORTED_TYPES.includes(column.type)) {
+        return false
+      }
+
+      return true
+    }
+
+    private isAutoColumn(column: OracleColumn) {
+      if (column.default && column.default.toLowerCase().includes("nextval")) {
+        return true
+      }
+
+      return false
+    }
+
+    /**
+     * Fetches the tables from the oracle table and assigns them to the datasource.
+     * @param {*} datasourceId - datasourceId to fetch
+     * @param entities - the tables that are to be built
+    */
+    async buildSchema(datasourceId: string, entities: Record<string, Table>) {
+      const columnsResponse = await this.internalQuery<ColumnsResponse>({ sql: this.COLUMNS_SQL })
+      const oracleTables = this.mapColumns(columnsResponse)
+
+      const tables: { [key: string]: Table } = {}
+
+      // iterate each table
+      Object.values(oracleTables).forEach(oracleTable => {
+        let table = tables[oracleTable.name]
+        if (!table) {
+          table = {
+            _id: buildExternalTableId(datasourceId, oracleTable.name),
+            primary: [],
+            name: oracleTable.name,
+            schema: {},
+          }
+          tables[oracleTable.name] = table
+        }
+
+        // iterate each column on the table
+        Object.values(oracleTable.columns)
+              // remove columns that we can't read / save
+              .filter(oracleColumn => this.isSupportedColumn(oracleColumn))
+              // match the order of the columns in the db
+              .sort((c1, c2) => c1.id - c2.id)
+              .forEach(oracleColumn => {
+          const columnName = oracleColumn.name
+          let fieldSchema = table.schema[columnName]
+          if (!fieldSchema) {
+            fieldSchema = {
+              autocolumn: this.isAutoColumn(oracleColumn),
+              name: columnName,
+              type: convertType(oracleColumn.type, TYPE_MAP),
```

Comment:
One thing I've been trying to do is just compress all of the SQL types into a single type map definition, as per the `convertSqlType` and `SQL_TYPE_MAP` in: https://github.com/Budibase/budibase/blob/a013c51e2bd27b69985e3b2d67147bfca81aafb7/packages/server/src/integrations/utils.ts#L10

Reply:
no worries! I will get this rebased from the latest changes in the mssql stuff, thanks for pointing that out